tokio/src/runtime/time/mod.rs

```diff
@@ -190,18 +190,22 @@ impl Driver {
         assert!(!handle.is_shutdown());
 
         // Finds out the min expiration time to park.
-        let expiration_time = (0..rt_handle.time().inner.get_shard_size())
-            .filter_map(|id| {
-                let lock = rt_handle.time().inner.lock_sharded_wheel(id);
-                lock.next_expiration_time()
-            })
+        let locks = (0..rt_handle.time().inner.get_shard_size())
+            .map(|id| rt_handle.time().inner.lock_sharded_wheel(id))
+            .collect::<Vec<_>>();
```

Comment:
It would be nice to avoid allocating this vector every time.

Reply:
Do you have any ideas to achieve this? Can we use `smallvec::SmallVec` here?

Reply:
We're not going to introduce a new dependency for it, but maybe a solution like that could work. But I don't think it has to be solved before we publish a release containing this fix.

(See also chat on discord.)